File: dot.net/Ogman.Internals/Ogman.Internals/PublicFacing.cs
using System.Text;

namespace Ogman.Internals
{
    public class PublicFacing : IInternalInterface
    {
        private readonly StringBuilder output;

        public PublicFacing(StringBuilder output)
        {
            this.output = output;
        }

        void IInternalInterface.DoInternalWork()
        {
            output.Append("internal work");
        }

        public void DoWork()
        {
            output.Append("Public work is based on ");
            IInternalInterface internalThis = ((IInternalInterface)this);
            internalThis.DoInternalWork();
            output.Append(".");
        }
    }
}

File: java/org.ogman.internals/src/org/ogman/internals/PublicFacing.java
package org.ogman.internals;

public class PublicFacing extends InternalInterface {

	private StringBuffer output;

	public PublicFacing(StringBuffer output) {
		this.output = output;
	}

	public void doWork() {
		output.append("Public work is based on ");
		doInternalWork();
		output.append(".");
	}

	@Override
	void doInternalWork() {
		output.append("internal work");
	}
}
